docs/examples/CanvasEvents.rst

```diff
@@ -0,0 +1,307 @@
+    the item that has been drawn needs to be referenced again.
+    Both of the following implementations are correct:
+
+    #. ID=app.Canvas1.create_oval(x1, y1, x2, y2, fill=python_green) 
+    #. app.Canvas1.create_oval(x1, y1, x2, y2, fill=python_green)
+    
+    In the first, the `ID`-identifier can (and in this example will) be used to reference the circle again. In the second implementation this is
+    impossible, but in many cases also not required.
+
+The `dotsfall`-function is the last to be defined. It is quite complex, so lets take it step by step.
+The following is the minimal-working function, although its effects are not entirely what we want.
+
+.. code-block:: python
+    
+    def dotsfall():
+        while not app.stop_event.is_set(): 
+            for j in app.listofdots:  
+                ID=j[0] 
+                speed=j[1]  
+                app.Canvas1.move(ID,0,speed) 
+                speed+=1  
+            app._root.after(50) 
+        app._root.update()
+        return
+
+The follwing is happening:
+
+#. A `while`-loop is created, which runs as long as the `app.stop_event` is not `set`.
+#. Within the `while`-loop, there is a `for`-loop, going over the `app.listofdots`
+#. Within the `for` loop, the `ID` of each Circle and the `speed` of each oval is read from the list
+#. The corresponding `Canvas1` item, the circles, are moved in Y+ direction (down)
+#. The speed is increased by one.
+#. After moving all objects on the `Canvas1`, the Threading event are temporarily stopped for 50ms, so that the GUI can be updated. Using `app._root` it is possible to directly address `TKinter` methods and functions
+#. After the `while` loop, the total GUI is forced to be updated using `app._root.update()`
+
+Some additional things need to happen during this function:
+
+#. The `app.listofdots` list needs to be updated after movement. For this, a temporary, local second list is introducted: `listofdots2`
+#. In case of a large number of dots, the `stop_event` needs to break the `for`-loop
+#. In case of a large number of dots, the `app._root.after()` needs to be shorter to keep the movement even.
+
+The updated, total function looks like this:
+
+.. code-block:: python
+    
+    def dotsfall():
+        listofdots2=[]  
+        while not app.stop_event.is_set():  ## continue movement until stop_event
+            for j in app.listofdots: 
+                if app.stop_event.is_set():
+                    break
+                ID=j[0]  
+                speed=j[1] 
+                app.Canvas1.move(ID,0,speed) 
+                speed+=1 
+                if app.Canvas1.coords(ID)[1]<800:
+                    listofdots2.append([ID,speed]) 
+            app.listofdots=listofdots2  
+            if len(app.listofdots)>100:   
+                waittime=int(max(10,100-(len(app.listofdots)/2)))  
+                app._root.after(waittime)  
+            else:
+                app._root.after(50) 
```

Comment:
Adjust here as well. Remember to change the explanations before this code block accordingly.